docs/tutorials/ruby-driver-crud-operations.txt

```diff
@@ -482,6 +470,173 @@ Deleting
+the level of acknowledgment requested from MongoDB for the particular write.
+More information about write concerns in general is available in the
+`MongoDB manual <https://docs.mongodb.com/manual/reference/write-concern/>`_.
+
+The Ruby driver supports specifying write concern on client, collection,
+session (for transactions on that session), transaction, GridFS bucket
+and write stream levels, as well as when manually issuing commands via
+``Database#command``.
+
+As of driver version 2.10, all driver objects accepting write concerns do so
+through the ``:write_concern`` option, which should be given a hash with
+the write concern options. Usage of the ``:write`` option is deprecated.
+In driver versions 2.9 and below, client, collection and GridFS objects
+took write concern options in the ``:write`` option with session and
+transaction objects employing the ``:write_concern`` option.
+
+Below are some examples of passing write concerns to client and colection
+objects. The ``:write_concern`` option can be provided when constructing
+new client and collection objects,  or to the ``#with`` methods.
+
+GridFS examples are provided on the :ref:`GridFS <gridfs>` page.
+
+.. code-block:: ruby
+
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write_concern: {w: 2})
+  alt_client = client.with(write_concern: {w: :majority})
+  
+  collection = client[:artists, write_concern: {w: 3}]
+  alt_collection = collection.with(write_concern: {w: :majority})
+  
+  # Uses w: 3
+  collection.insert_one({name: 'SUN Project'})
+  # Uses w: :majority
+  alt_collection.insert_one({name: 'SUN Project'})
+
+Driver versions 2.9 and earlier accepted write concerns on client and collection
+level via the ``:write`` option. This usage continues to be supported for
+backwards compatibility, but is deprecated:
+
+.. code-block:: ruby
+
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write: {w: 2})
+  alt_client = client.with(write: {w: :majority})
+  
+  collection = client[:artists, write: {w: 3}]
+  alt_collection = collection.with(write: {w: :majority})
+
+If both ``:write`` and ``:write_concern`` options are provided, their
+values must be identical or an exception will be raised:
+
+.. code-block:: ruby
+
+  # OK
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write_concern: {w: 3}, write: {w: 3})
+  
+  # Error
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write_concern: {w: 3}, write: {w: :majority})
+
+When ``#with`` methods are used to alter the options on a client or collection,
+the last provided option wins in case of naming differences:
+
+.. code-block:: ruby
+
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write_concern: {w: 2})
+  alt_client = client.with(write: {w: 3})
+  alt_client.options[:write]
+  alt_client.options[:write_concern]
```

Comment:
Should this include what the resulting values are for `alt_client.options[:write]` and `alt_client.options[:write_concern]`?

Reply:
Yes indeed, thank you.